main.go

```diff
@@ -62,61 +59,26 @@ func main() {
 }
 
 func dbOptions(environment string) (postgres.Options, error) {
-	if strings.ToLower(environment) == "staging" {
-		rawConnectionURL, valid := os.LookupEnv("POSTGRES_URL")
-		if !valid {
-			return postgres.Options{}, fmt.Errorf("staging missing valid POSTGRES_URL for PostgreSQL connection")
-		}
-
-		connectionURL, err := url.Parse(rawConnectionURL)
-		if err != nil {
-			return postgres.Options{}, fmt.Errorf("staging environment variable POSTGRES_URL does not contain valid connection info")
-		}
-
-		user := connectionURL.User.Username()
-		password, _ := connectionURL.User.Password()
-		host, portStr, err := net.SplitHostPort(connectionURL.Host)
-		dbName := connectionURL.Path[1:]
-		sslMode, validSSLMode := os.LookupEnv("POSTGRES_SSL")
-		if !validSSLMode {
-			return postgres.Options{}, fmt.Errorf("staging missing environment variable POSTGRES_SSL specifying SSL mode")
-		}
-
-		port, err := strconv.Atoi(portStr)
-		if err != nil {
-			return postgres.Options{}, fmt.Errorf("staging POSTGRES_URL must contain valid port number")
-		}
-
-		maxConnections, err := getMaxConnections()
-		if err != nil {
-			return postgres.Options{}, err
-		}
-
-		return postgres.Options{User: user, Pass: password, Host: host, Port: port, DBName: dbName, SSLMode: sslMode, MaxConnections: maxConnections, StatementTimeout: 5000}, nil
-	} else if strings.ToLower(environment) == "production" {
-		user, validUser := os.LookupEnv("POSTGRES_USER")
-		password, validPass := os.LookupEnv("POSTGRES_PASSWORD")
-		host, validHost := os.LookupEnv("DATABASE_URL")
-		portStr, validPort := os.LookupEnv("POSTGRES_PORT")
-		dbName, validDBName := os.LookupEnv("POSTGRES_DB")
-		sslMode, validSSLMode := os.LookupEnv("POSTGRES_SSL")
-
-		if !validUser || !validPass || !validHost || !validPort || !validDBName || !validSSLMode {
-			return postgres.Options{}, fmt.Errorf("production environment missing required environment variables for PostgreSQL database")
-		}
-		port, err := strconv.Atoi(portStr)
-		if err != nil {
-			return postgres.Options{}, fmt.Errorf("production environment must contain valid 'POSTGRES_PORT' number")
-		}
-
-		maxConnections, err := getMaxConnections()
-		if err != nil {
-			return postgres.Options{}, err
-		}
+	maxConnections, err := getMaxConnections()
+	if err != nil {
+		return postgres.Options{}, err
+	}
 
-		return postgres.Options{User: user, Pass: password, Host: host, Port: port, DBName: dbName, SSLMode: sslMode, MaxConnections: maxConnections, StatementTimeout: 5000}, nil
+	port, err := strconv.Atoi(os.Getenv("POSTGRES_PORT"))
```

Comment:
Using `os.Getenv` rather than `os.LookupEnv` makes the code a bit more concise but is going to make hunting for a miss-typed environment variable more painful. If you're worried about boilerplate you could make a helper function to return an environment variable or a nice error message. Up to you though, if you feel slimming down the coded is worth it go for it.

Reply:
I feel like it's worth it, we're not going to mistype environment variable names in production, and even if we did, we would just get a message telling us we couldn't connect on port 0 (if we mistyped `POSTGRES_PORT`) which would make it obvious that we messed up our environment variables. In addition, I like the "zero value" idea that os.Getenv uses instead of an error. If someone doesn't specify POSTGRES_USER or POSTGRES_PORT maybe they just mean just don't want to specify them! Not necessarily that they forgot to.